src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -25,12 +25,12 @@ public class TagHelperContext
         /// <param name="getChildContentAsync">A delegate used to execute and retrieve the rendered child content 
         /// asynchronously.</param>
         public TagHelperContext(
-            [NotNull] IDictionary<string, object> allAttributes,
+            [NotNull] IEnumerable<IReadOnlyTagHelperAttribute> allAttributes,
             [NotNull] IDictionary<object, object> items,
             [NotNull] string uniqueId,
             [NotNull] Func<Task<TagHelperContent>> getChildContentAsync)
         {
-            AllAttributes = allAttributes;
+            AllAttributes = new ReadOnlyTagHelperAttributes<IReadOnlyTagHelperAttribute>(allAttributes);
```

Comment:
need to perform a deep copy of the `TagHelperAttribute` (aka `IReadOnlyTagHelperAttribute`) members.  the collection is not currently guaranteed to reflect the original element's attributes.